activesupport/lib/active_support/cache/strategy/local_cache.rb

```diff
@@ -129,17 +129,23 @@ def read_multi_entries(names, **options)
             keys_to_names = names.index_by { |name| normalize_key(name, options) }
 
             local_entries = local_cache.read_multi_entries(keys_to_names.keys)
-            local_entries.transform_keys! { |key| keys_to_names[key] }
-            local_entries.transform_values! do |payload|
-              deserialize_entry(payload, **options)&.value
-            end
-            missed_names = names - local_entries.keys
 
-            if missed_names.any?
-              local_entries.merge!(super(missed_names, **options))
-            else
-              local_entries
+            results = local_entries.each_with_object({}) do |(key, value), result|
+              normalized_key = keys_to_names[key]
+              version = normalize_version(normalized_key, options)
+
+              entry = deserialize_entry(value, **options)
+              if entry && (entry.expired? || entry.mismatched?(version))
```

Comment:
```suggestion
              if entry && (entry.expired? || entry.mismatched?(normalize_version(keys_to_names[key], options)))
```

Would be better, because while more verbose we'd save on computing the version for missing and expired entries.

Reply:
Also if `entry` is `nil` we should skip.

Reply:
Nevermind, that's part of the spec.